lib/JSON/Validator.pm

```diff
@@ -1290,7 +1290,10 @@ See L</Bundled specifications> for more details.
   my $jv = $jv->formats(\%hash);
 
 Holds a hash-ref, where the keys are supported JSON type "formats", and
-the values holds a code block which can validate a given format. A code
+the values holds a code block which can validate a value against a given
+format. The second parameter is the schema fragment where the format is
+defined, in case some custom format requires accessing additional schema
+attributes in order to tweak its validation. A code
```

Comment:
I think this documentation is very confusing now. I suggest trying to illustrate how it works with a better code example. Something like this:

```
=head2 formats
  
  my $hash_ref = $jv->formats;
  my $jv       = $jv->formats(\%hash);
  
Holds a hash-ref, where the keys are supported JSON type "formats", and the 
values holds a code block which can validate a given format. The code block
must return C<undef> on success and a string on error:
  
  $jv->formats->{custom_format} = sub {
    my $value                   = shift; # User input
    my $schema_fragment         = shift; # Example: {type => "string", format => "custom_format"}
    my $path_to_schema_fragment = shift; # Example "/some/location/in/schema"
  
    return defined $value && $value eq "42" ? undef : "Not 42.";
  };  
  
See L<JSON::Validator::Formats> for a list of supported formats.
```